src/components/policy/policy_external/src/policy_helper.cc

```diff
@@ -342,6 +359,8 @@ bool CheckAppPolicy::operator()(const AppPoliciesValueType& app_policy) {
   if (!IsPredefinedApp(app_policy)) {
     SetPendingPermissions(app_policy, result);
     AddResult(app_id, result);
+  } else {
+    AddRelatedAppsResult(app_id, result);
```

Comment:
@AKalinich-Luxoft 
This part is inside bool operator()(<application_policy>&) which basically iterates over all applications policies. Why you're iterating over all applications policies (AddRelatedAppsResult call) inside another iteration over same data?
Why you can't reuse  AddResult()?

Reply:
@dev-gh because in operator() we check permission changes only for current app_id. So in case when you will receive PTU where only "default" policy was changed, you will get out_results like:
default: RESULT_CONSENT_NOT_REQIURED
However if application with App1 was registered and using default policies, it will not receive OnPermissionChange notification later in ProcessAppPolicyCheckResults(), because its permissions were changed indirectly.
This AddRelatedAppsResult performs additional iteration over all applications in case some of predefined application policies permissions were changed, in order to add applications, which permissions were changed indirectly, i.e:
default: RESULT_CONSENT_NOT_REQIURED
App1: RESULT_CONSENT_NOT_REQIURED
In this case App1 will receive OnPermissionChange notification as expected.

Reply:
@AKalinich-Luxoft 
As I responded in my previous comment I suspect there is an issue with the detecting of 'default' groups changes and further results processing. So I think there is no need for additional checks but instead, we have to properly detect changes of 'default' section before applying of PTU, put correct result for that and send the notification. I'll try to check it on my side and let you know.

Reply:
@dev-gh I suppose we have to add new default permissions for App1 to PTU (if it is absent there) before its applying? In this case all this sequence should work correctly.

Reply:
@dev-gh not actual after 89ceac249b54ec02c042865b27f4af8b2363d450